### DISCUSSION TITLE
How to delete multiples files in Github

### DISCUSSION THREAD

**Discussion by naqvisn** [community/community#22543] [New to GitHub]
<p>Is there a way to select all files and delete files in github at once? Currently I delete single file at time and commit changes.</p>
<p>I want to select multiple or all files and delete under a single commit.</p>
<p>Thanks</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/naqvisn">@naqvisn</a>,</p>
<p>Thanks for being part of the GitHub Community Forum!</p>
<p>Your post was edited because it contained potentially sensitive information that you may not have intended to make public, such as your email address shared as a tag. We discourage users from sharing such information in our public forum, as it can create security and privacy risks for the poster, as well as other users.</p>
<p>If you want to share a repo with others in the Community Forum, we recommend sharing a link to the GitHub repo in which the project you are working on exists, rather than sharing an email address. </p>
<p>Cheers!</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/naqvisn">@naqvisn</a>,</p>
<p>There isn’t a way to select multiple files for deletion using the Web UI at the moment, but I’ll let the team know you’d like us to add this feature in the future.</p>
<p>In the meantime, the way we recommend editing or deleting multiple files is by working with a local repository. You would then be able to delete the files in your local clone, commit that change to your local repository, and then push that change to the remote repository on GitHub.</p>
<p>The steps for doing this are:</p>
<ul>
<li>In the command-line, navigate to your local repository.</li>
<li>Ensure you are in the default branch:<br>
<code>git checkout master</code>
</li>
<li>The <code>rm -r</code> command will recursively remove your folder:<br>
<code>git rm -r folder-name</code>
</li>
<li>Commit the change:<br>
<code>git commit -m "Remove duplicated directory"</code>
</li>
<li>Push the change to your remote repository:<br>
<code>git push origin master</code>
</li>
</ul>
<p>I hope this helps!</p>

**Reply by PhilMinecraft** [reply]
Why need a command line when we can just remotely access and delete select files in bulk using vs code? I'll just install [Remote Repositories](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-repositories), sign in my GitHub account there, select a repo and I can delete select files in bulk then commit the change (aka save changes). This method is much easier and less hassle than your method. :/

**Reply by y01cu** [reply]
@PhilMinecraft thanks for sharing that plugin here, it helped me.

**Reply by PhilMinecraft** [reply]
This should have been accepted before

**Comment by AstroSpaceDiscord** [comment]
<p>how do i open the command line?</p>

**Comment by mabagheri** [comment]
<p>One easy way: download and install git from “<a href="https://git-scm.com/download/win" rel="nofollow noopener">https://git-scm.com/download/win</a>”. In windows explorer, go to the folder you want, right-click and run “Git bash here”. </p>

**Comment by kondareddy1229** [comment]
<p>Thank you brother</p>

**Comment by mghhimu** [comment]
<p>press Ctl+ALT+t to open a new terminal in Ubuntu</p>

**Comment by chuankin** [comment]
<p>How do I delete multiple files using github desktop?</p>

**Comment by Jadamoureen** [comment]
<p>Thank you so much …</p>

**Comment by realleyriley** [comment]
<p>For those looking for a git solution (command line solution):</p>
<ol>
<li>
<p>Manually delete the files just like you would normally delete any file in windows. Right click the file(s) and click delete. </p>
</li>
<li>
<p>Type this command</p>
<p>git add -u</p>
</li>
</ol>
<p>You may go ahead and commit the change. </p>

**Reply by adithya-s-sekhar** [reply]
Thank you this helped me with renaming multiple files. First tracked the deleted files with `git add -u` and then tracked the new files with `git add *`, it showed up as renamed in `git status`.

**Comment by PhakaSosibo** [comment]
<p>Thank you somuch</p>

**Reply by Effionggodwinnsikan** [reply]
am caught in the middle of a bad collaborator and want to push my code to a different repo

**Comment by avDec25** [comment]
<p>You may need to use:</p>
<pre><code>git push -f origin master
</code></pre>
<p>How it worked for me:</p>
<pre><code>➜ mfb-plm git:(master) git push origin master
To bitbucket.org:myntra/mfb-plm.git
 ! [rejected] master -&gt; master (fetch first)
error: failed to push some refs to 'git@bitbucket.org:myntra/mfb-plm.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
➜ mfb-plm git:(master) **git push -f origin master** Enumerating objects: 383, done.
Counting objects: 100% (383/383), done.
Delta compression using up to 12 threads
Compressing objects: 100% (94/94), done.
Writing objects: 100% (383/383), 241.06 KiB | 80.35 MiB/s, done.
Total 383 (delta 238), reused 381 (delta 238)
remote: Resolving deltas: 100% (238/238), done.
To bitbucket.org:myntra/mfb-plm.git
 + a4a0df9...849eece master -&gt; master (forced update)
</code></pre>

**Comment by Logeswaranbca** [comment]
<p>Its deleting folder i want to delete inside folder file’s</p>

**Comment by DocLock17** [comment]
<p>One Solution would be:</p>
<p>  1. “View” &gt;&gt; “Show in Explorer”.</p>
<p>  2. Make edits in your file explorer, and close.</p>
<p>  3. Commit Changes and/or Pull and Merge.</p>

**Comment by Ajithkumar-Hariraman** [comment]
<p>Thanks. It works.</p>

**Comment by Mona145** [comment]
<p>Thanks. Appreciate the help.</p>

**Comment by prady1900** [comment]
<p>It actually deleted the read.md file too</p>

**Comment by ghost** [comment]
<p>thanks. But can you add these feautere to web UI? because this id going to be a simple way.</p>

**Comment by VoliaSi** [comment]
<p>I get “fatal: pathspec … did not mach any files”, though files are added to GitHub when I add another directory.</p>
<p>I was studying to work ith git with GitBush, GitGUI, Intellij IDEA, TortoiseGit and, think, made some modifications, which I cannot get rid of now.<br>
Any commit adds too many extra folders, which I can’t delete from git with your instructions.</p>

**Comment by NitinIsFit** [comment]
I have  few questions in my mind too, "how to move multiple files from one folder to another in a single commit?", and "how to rename a folder or image?"

**Comment by NitinIsFit** [comment]
Wai- wha-, this question is 4 years old!

**Comment by arjunj132** [comment]
I'm a bit late, but another way you could do it is that you could click `.` and then the `github.dev` vscode interface will popup. Then you could just select all the files and then delete then through right-click and click delete. Then you could just commit it on the `Source Control` tab. This way is a bit faster than using the command line.

**Comment by Rorararara** [comment]
Why there is no feature in 2026 to delete multiple files without command lines ?
